doc/user_guide/transform/aggregate.rst

```diff
@@ -58,7 +58,96 @@ The same plot can be shown using an explicitly computed aggregation, using the
+      cars_df.groupby('Cylinders')
+      .Acceleration
+      .mean()
+      .reset_index()
+      .rename(columns={'Acceleration': 'mean_acc'})
+   )
+
+   alt.Chart(source).mark_bar().encode(
+      y='Cylinders:O',
+      x='mean_acc:Q'
+   )
+
+**Note:** As mentioned in :doc:`../data`, this approach of transforming the
+data with Pandas is preferable if we already have the DataFrame at hand.
+
+Because :code:`Cylinders` is of type :code:`int64` in the :code:`source`
+DataFrame, Altair would have treated it as a :code:`qualitative` --instead of
+:code:`ordinal`-- type, had we not specified it. Making the type of data
+explicit is important since it affects the resulting plot; see
+:ref:`type-legend-scale` and :ref:`type-axis-scale` for two illustrated
+examples. As a rule of thumb, it is better to make the data type explicit,
+instead of relying on an implicit type conversion.
+
+Functions Without Arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is possible for aggregate functions to not
+have an argument. In this case, aggregation will be performed on the column
+used in the other axis.
+
+The following chart demonstrates this by counting the number of cars with
+respect to their country of origin.
+
+.. altair-plot::
+
+   alt.Chart(cars).mark_bar().encode(
+      y='Origin:N',
+      # shorthand form of alt.Y(aggregate='count')
+      x='count()'
+   )
+
+**Note:** The :code:`count` aggregate function is of type
+:code:`quantitative` by default, it does not matter if the source data is a
+DataFrame, URL pointer, CSV file or JSON file.
+
+Functions that handle categorical data (such as :code:`count`,
+:code:`missing`, :code:`distinct` and :code:`valid`) are the ones that get
+the most out of this feature.
+
+Argmin / Argmax
+^^^^^^^^^^^^^^^
+Both :code:`argmin` and :code:`argmax` aggregate functions can only be used
+with the :meth:`~Chart.transform_aggregate` method. Trying to use their
+respective shorthand notations will result in an error. This is due to the fact
+that either :code:`argmin` or :code:`argmax` functions return an object, not
+values.  This object then specifies the values to be selected from other
+columns when encoding.  One can think of the returned object as being a
+dictionary, while the column serves the purpose of being a key, which then
+obtains its respective value.
+
+The true value of these functions is appreciated when we want to compare the
+most **distinctive** samples from two sets of data with respect to another set
+of data.
+
+As an example, suppose we want to compare the weight of the strongest cars,
+with respect to their country/region of origin. This can be done using
+:code:`argmax`:
+
+.. altair-plot::
+
+   alt.Chart(cars).mark_bar().encode(
+      x='greatest_hp[Weight_in_lbs]:Q',
+      y='Origin:N'
+   ).transform_aggregate(
+      greatest_hp='argmax(Horsepower)',
+      groupby=['Origin']
+   )
+
+It is clear that Japan's strongest car is also the lightest, while that of USA
+is the heaviest.
```

Comment:
```suggestion
The :code:`argmin` and :code:`argmax` functions help you find values from
one field that correspond to the minimum or maximum values in another
field. For example, you might want to find the production budget of
movies that earned the highest gross revenue in each genre.

These functions must be used with the :meth:`~Chart.transform_aggregate`
method rather than their shorthand notations. They return objects that act
as selectors for values in other columns, rather than returning values
directly. You can think of the returned object as a dictionary where the
column serves as a key to retrieve corresponding values.


To illustrate this, let's compare the weights of cars with the highest
horsepower across different regions of origin:

.. altair-plot::

   alt.Chart(cars).mark_bar().encode(
      x='greatest_hp[Weight_in_lbs]:Q',
      y='Origin:N'
   ).transform_aggregate(
      greatest_hp='argmax(Horsepower)',
      groupby=['Origin']
   )

This visualization reveals an interesting contrast: among cars with the
highest horsepower in their respective regions, Japanese cars are notably
lighter, while American cars are substantially heavier.
```